src/updater/updater.cpp

```diff
@@ -149,446 +95,569 @@ void CheckWeightsConsistency(
+    storage::io::FileReader nodes_file(config.node_based_graph_path,
+                                       storage::io::FileReader::HasNoFingerprint);
+    nodes_file.DeserializeVector(internal_to_external_node_map);
+
+    // vector to count used speeds for logging
+    // size offset by one since index 0 is used for speeds not from external file
+    using counters_type = std::vector<std::size_t>;
+    std::size_t num_counters = config.segment_speed_lookup_paths.size() + 1;
+    tbb::enumerable_thread_specific<counters_type> segment_speeds_counters(
+        counters_type(num_counters, 0));
+    const constexpr auto LUA_SOURCE = 0;
+
+    // The check here is enabled by the `--edge-weight-updates-over-factor` flag it logs a
+    // warning if the new duration exceeds a heuristic of what a reasonable duration update is
+    std::unique_ptr<extractor::SegmentDataContainer> segment_data_backup;
+    if (config.log_edge_updates_factor > 0)
+    {
+        // copy the old data so we can compare later
+        segment_data_backup = std::make_unique<extractor::SegmentDataContainer>(segment_data);
+    }
 
-    auto mmap_file = [](const std::string &filename, boost::interprocess::mode_t mode) {
-        using boost::interprocess::file_mapping;
-        using boost::interprocess::mapped_region;
+    tbb::concurrent_vector<GeometryID> updated_segments;
 
-        try
+    using DirectionalGeometryID = extractor::SegmentDataContainer::DirectionalGeometryID;
+    auto range = tbb::blocked_range<DirectionalGeometryID>(0, segment_data.GetNumberOfGeometries());
+    tbb::parallel_for(range, [&, LUA_SOURCE](const auto &range) {
+        auto &counters = segment_speeds_counters.local();
+        std::vector<double> segment_lengths;
+        for (auto geometry_id = range.begin(); geometry_id < range.end(); geometry_id++)
         {
-            const file_mapping mapping{filename.c_str(), mode};
-            mapped_region region{mapping, mode};
-            region.advise(mapped_region::advice_sequential);
-            return region;
+            auto nodes_range = segment_data.GetForwardGeometry(geometry_id);
+
+            segment_lengths.clear();
+            segment_lengths.reserve(nodes_range.size() + 1);
+            util::for_each_pair(nodes_range, [&](const auto &u, const auto &v) {
+                segment_lengths.push_back(util::coordinate_calculation::greatCircleDistance(
+                    util::Coordinate{internal_to_external_node_map[u]},
+                    util::Coordinate{internal_to_external_node_map[v]}));
+            });
+
+            auto fwd_weights_range = segment_data.GetForwardWeights(geometry_id);
+            auto fwd_durations_range = segment_data.GetForwardDurations(geometry_id);
+            auto fwd_datasources_range = segment_data.GetForwardDatasources(geometry_id);
+            bool fwd_was_updated = false;
+            for (const auto segment_offset : util::irange(0, fwd_weights_range.size()))
```

Comment:
SFINAE fails for (int, size_t) -> `util::irange<std::size_t>(0, fwd_weights_range.size()))`